semantic-python/src/Prelude.score

```diff
@@ -1,8 +1,22 @@
 {
-  type <- \name -> \bases -> \dict ->
-    #record { __name: name, __bases: bases, __dict: dict };
+  type <- \name -> \super -> \slots ->
+    #record { __name: name, __super: super, __slots: slots };
 
-  object <- type "object" #unit #record{};
+  instance <- \class -> \prim -> \slots ->
+    #record { __super: class, __prim: prim, __slots: slots };
 
-  #record { type: type, object: object }
+  -- object's superclass is type itself
+  object <- type "object" type #record{};
+
+  str <- type "str" object #record { };
+  str.__new__ = (\contents -> instance str contents #record{});
+
+  getitem <- \super -> \item -> \attr ->
+    if item.slots.?attr then item.slots.attr else #unit;
```

Comment:
This should be

```
  getitem <- rec getitem = \item -> \attr ->
    if item.slots.?attr 
    then item.slots.attr 
    else if item.?class
    then getitem (item.class) attr
    else #unit;
```

but the scope graphing system doesn’t handle `rec` constructs correctly (#362), so we can’t recursively call `getitem`. This is fine for now.